client/pages/me/me.js

```diff
@@ -1,66 +1,71 @@
-// pages/me/me.js
-Page({
+//index.js
+const app = getApp()
 
-  /**
-   * 页面的初始数据
-   */
+Page({
   data: {
-
-  },
-
-  /**
-   * 生命周期函数--监听页面加载
-   */
-  onLoad: function (options) {
-
+    userInfo: {},
+    chName: null,
+    chNameTemp: null,
+    egName: null,
+    egNameTemp: null,
+    phone: null,
+    phoneTemp: null,
+    email: null,
+    emailTemp: null,
+
+    saveBtnType: 'default'
   },
 
-  /**
-   * 生命周期函数--监听页面初次渲染完成
-   */
-  onReady: function () {
-
-  },
 
-  /**
-   * 生命周期函数--监听页面显示
-   */
-  onShow: function () {
+  onLoad: function () {
+    console.log(app.userInfo)
+    this.setData({
+      userInfo: app.userInfo
+    })
 
+    if (this.data.userInfo == null) {
+        // if there is no userinfo yet
+        setTimeout(function () {}, 100)
+    } 
   },
+ 
 
-  /**
-   * 生命周期函数--监听页面隐藏
-   */
-  onHide: function () {
-
+  chNameInput: function(event) {
+    this.setData({
+      chNameTemp: event.detail.value,
+      saveBtnType: 'primary'
+    })
   },
-
-  /**
-   * 生命周期函数--监听页面卸载
-   */
-  onUnload: function () {
-
+  egNameInput: function (event) {
+    this.setData({
+      egNameTemp: event.detail.value,
+      saveBtnType: 'primary'
+    })
   },
-
-  /**
-   * 页面相关事件处理函数--监听用户下拉动作
-   */
-  onPullDownRefresh: function () {
-
+  phoneInput: function (event) {
+    this.setData({
+      phoneTemp: event.detail.value,
+      saveBtnType: 'primary'
+    })
   },
-
-  /**
-   * 页面上拉触底事件的处理函数
-   */
-  onReachBottom: function () {
-
+  emailInput: function (event) {
+    this.setData({
+      emailTemp: event.detail.value,
+      saveBtnType: 'primary'
+    })
   },
 
-  /**
-   * 用户点击右上角分享
-   */
-  onShareAppMessage: function () {
+  infoSaveClick: function(event) {
+    this.setData({
```

Comment:
I think we should update the user info on server via the `POST` endpoint.

Reply:
And probably we should ignore the button click event if `saveBtnType  === 'default'`.

Reply:
Should we still post the data even if btnDisable is set to true?